File: app/src/main/java/pozzo/apps/travelweather/ui/fragment/SideMenuFragment.java
package pozzo.apps.travelweather.ui.fragment;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import pozzo.apps.travelweather.R;

/**
 * Sera nosso menu lateral =D.
 *
 * Created by sarge on 10/29/15.
 */
public class SideMenuFragment extends Fragment {
	//TODO eh mesmo uma boa ideia salvar o id? Eu acho que nao...
	private RadioGroup rgDaySelection;

	@Nullable
	@Override
	public View onCreateView(
			LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		View contentView = inflater.inflate(R.layout.fragment_side_menu, container, false);

		rgDaySelection = (RadioGroup) contentView.findViewById(R.id.rgDaySelection);

		fillView(contentView);

		return contentView;
	}

	private void fillView(View contentView) {
		SharedPreferences preferences =
				PreferenceManager.getDefaultSharedPreferences(getActivity());
		int selectedDay = preferences.getInt("selectedDay", R.id.rToday);
		RadioButton selectedRadio = (RadioButton) contentView.findViewById(selectedDay);
		selectedRadio.setChecked(true);

		rgDaySelection.setOnCheckedChangeListener(onDaySelection);
	}

	private RadioGroup.OnCheckedChangeListener onDaySelection =
			new RadioGroup.OnCheckedChangeListener() {
		@Override
		public void onCheckedChanged(RadioGroup group, int checkedId) {
			saveSelection();
		}
	};

	private void saveSelection() {
		SharedPreferences.Editor preferences =
				PreferenceManager.getDefaultSharedPreferences(getActivity()).edit();

		int selectedId = rgDaySelection.getCheckedRadioButtonId();
		if(selectedId > 0)
			preferences.putInt("selectedDay", selectedId).apply();
	}
}

File: app/src/main/java/pozzo/apps/travelweather/map/ui/SideMenuFragment.kt
package pozzo.apps.travelweather.map.ui

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RadioButton
import android.widget.RadioGroup
import pozzo.apps.travelweather.R
import pozzo.apps.travelweather.databinding.FragmentSideMenuBinding
import pozzo.apps.travelweather.map.viewmodel.MapViewModel
import pozzo.apps.travelweather.map.viewmodel.PreferencesViewModel

/**
 * Our main menu.
 */
class SideMenuFragment : Fragment() {
  private lateinit var preferencesViewModel: PreferencesViewModel
  private lateinit var viewModel: MapViewModel

  override fun onAttach(context: Context?) {
    super.onAttach(context)
    preferencesViewModel = ViewModelProviders.of(activity!!).get(PreferencesViewModel::class.java)
    viewModel = ViewModelProviders.of(activity!!).get(MapViewModel::class.java)
  }

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    val contentView = DataBindingUtil.inflate<FragmentSideMenuBinding>(inflater, R.layout.fragment_side_menu, container, false)
    contentView.viewModel = viewModel
    setupView(contentView.root)
    return contentView.root
  }

  /**
   * Fill view components with data.
   */
  private fun setupView(contentView: View) {
    preferencesViewModel.selectedDay.observe(this, Observer { selectedDate ->
      val selectedRadio = contentView.findViewById<RadioButton>(selectedDate!!.resourceId)
      selectedRadio.isChecked = true
    })

    contentView.findViewById<RadioGroup>(R.id.rgDaySelection)
        .setOnCheckedChangeListener({ _, checkedId -> preferencesViewModel.setSelectedDay(checkedId) })
  }
}
